### PR TITLE
gh: Upgrade to 2.62.0 to address HIGH CVE-2024-52308; add generate-tarball.sh script for auto-upgrading support

### PR DESCRIPTION
**Pull request by vinceaperri**
###### Merge Checklist  

**All** boxes should be checked before merging the PR *(just tick any boxes which don't apply to this PR)*
- [x] The toolchain has been rebuilt successfully (or no changes were made to it)
- [x] The toolchain/worker package manifests are up-to-date
- [x] Any updated packages successfully build (or no packages were changed)
- [x] Packages depending on static components modified in this PR (Golang, `*-static` subpackages, etc.) have had their `Release` tag incremented.
- [x] Package tests (%check section) have been verified with RUN_CHECK=y for existing SPEC files, or added to new SPEC files
- [x] All package sources are available
- [x] cgmanifest files are up-to-date and sorted (`./cgmanifest.json`, `./toolkit/scripts/toolchain/cgmanifest.json`, `.github/workflows/cgmanifest.json`)
- [x] LICENSE-MAP files are up-to-date (`./LICENSES-AND-NOTICES/SPECS/data/licenses.json`, `./LICENSES-AND-NOTICES/SPECS/LICENSES-MAP.md`, `./LICENSES-AND-NOTICES/SPECS/LICENSE-EXCEPTIONS.PHOTON`)
- [x] All source files have up-to-date hashes in the `*.signatures.json` files
- [x] `sudo make go-tidy-all` and `sudo make go-test-coverage` pass
- [x] Documentation has been updated to match any changes to the build system
- [x] Ready to merge

---

###### Summary 

Upgrades gh to 2.62.0 to address CVE-2024-52308

###### Change Log  

- Upgrade gh to 2.62.0
- Add gh patch to fix false negative when running tests as root
- Add helper script to generate gh source and vendor tarballs and update signatures

###### Does this affect the toolchain?  

**NO**

###### Links to CVEs  
- https://nvd.nist.gov/vuln/detail/CVE-2024-52308

###### Test Methodology

- Pipeline build id: 679171

### CODE REVIEW

File: SPECS/gh/generate-tarballs.sh
Comment (mbykhovtsev-ms) [inline SPECS/gh/generate-tarballs.sh:1]: I believe our automation might be expective certain arguments to be passed such as `srcTarball`, `outFolder` and `pkgVersion`, I believe the file might also need to be called `generate-tarball.sh`. Would you be able to confirm that with Nicolas Guibourge?

Comment (nicogbg) [reply SPECS/gh/generate-tarballs.sh:1]: Mykhailo is right about your comment (see https://github.com/microsoft/azurelinux/blob/3.0/SPECS/coredns/generate_source_tarball.sh for example)
Adding the params and taking them in to account will let auto-upgrade work for that spec

**Review by mbykhovtsev-ms — CHANGES_REQUESTED** [review]
The `generate_tarball.sh` needs to be fixed

File: SPECS/gh/generate-tarballs.sh
Comment (mbykhovtsev-ms) [reply SPECS/gh/generate-tarballs.sh:1]: Thanks, one more change, we need the script to be named `generate-tarball.sh` (remove the "s" at the end)

**Review by mbykhovtsev-ms — APPROVED** [review]

**Review by jslobodzian — APPROVED** [review]

**Pull request merged**